### PR TITLE
MGDCTRS-1667: Update the Namespace card information

### PR DESCRIPTION
**Pull request by indraraj**
Updated the Namespace card content with the latest feedback.
- Preview Namespace card will display the `Owner` and `Cluster Id` but no `Cluster name`
- Customer OSD Namespace cards will display both `Cluster ID` and `Cluster name` but no `Owner`
- For the `Cluster name` loading state loading Skelton is displayed in place of the loading spinner.
- Using the new [selectable PF Card](https://www.patternfly.org/v4/components/card#selectable-cards) component 
- Non-ready cards are now grayed out and can not be hovered or selected.

Moved away from calling cluster API for each Namespace card to a single call fetching all the cluster lists.
Screenshot
image
Loading state
image
In case of Cluster API result in an error(allowing the user to continue with _"Could not get the cluster name"_ msg in place of Cluster name as its non-blocking info)
image

### CODE REVIEW

**Comment by indraraj** [comment]
Left the `Cluster Id` on the eval card as we search by `Cluster id`
image

File: src/app/pages/CreateConnectorPage/StepNamespace.tsx [hunk 1]
```diff
@@ -94,10 +107,58 @@ const ClustersGallery: FunctionComponent = () => {
...
+    const responseCopyItems = { ...response }.items!.map((item) => {
+      return {
+        ...item,
+        ...(!item.expiration && {
+          clusterName: t('clusterError'),
+        }),
+      };
+    });
+    setUpdateResponse({ ...response!, items: responseCopyItems });
+  }, [response]);
```
Comment (gashcrumb) [inline src/app/pages/CreateConnectorPage/StepNamespace.tsx:146]: The lint warning here is probably worth fixing.

File: src/app/pages/CreateConnectorPage/StepNamespace.tsx [hunk 2]
```diff
@@ -94,10 +107,58 @@ const ClustersGallery: FunctionComponent = () => {
...
+
+  useEffect(() => {
+    if (!isEqual(prevResponse, response)) {
+      setUpdateResponse(cloneDeep(response));
+      getClusters({
+        accessToken: getToken,
+        connectorsApiBasePath: connectorsApiBasePath,
+      })(getClusterInfo, onClusterError);
+    }
+  }, [connectorsApiBasePath, getClusterInfo, getToken, prevResponse, response]);
```
Comment (gashcrumb) [inline src/app/pages/CreateConnectorPage/StepNamespace.tsx:156]: This lint warning too probably.

**Comment by gashcrumb** [comment]
Tested locally, I really like the updated cards.  So just needs the lint warnings sorted.

File: src/app/pages/CreateConnectorPage/StepNamespace.tsx (hunk 1 above)
Comment (indraraj) [reply src/app/pages/CreateConnectorPage/StepNamespace.tsx]: Done

File: src/app/pages/CreateConnectorPage/StepNamespace.tsx (hunk 2 above)
Comment (indraraj) [reply src/app/pages/CreateConnectorPage/StepNamespace.tsx]: Done

**Comment by indraraj** [comment]
> Tested locally, I really like the updated cards. So just needs the lint warnings sorted.

Sure 👍  don't know why my lint plugging didn't complain about these.

**Comment by aprilma419** [comment]
Thanks @indraraj for the updates, Looking good to me.  I think my only question is from these current cards, it's a little hard to figure out which one is the preview namespace at a glance, do we want to add the labels for now？

Screen Shot 2022-11-30 at 17 41 28

**Comment by indraraj** [comment]
> Thanks @indraraj for the updates, Looking good to me. I think my only question is from these current cards, it's a little hard to figure out which one is the preview namespace at a glance, do we want to add the labels for now？
> 
> Screen Shot 2022-11-30 at 17 41 28

sure I will update it 👍

**Comment by gashcrumb** [comment]
> > Tested locally, I really like the updated cards. So just needs the lint warnings sorted.
> 
> Sure +1 don't know why my lint plugging didn't complain about these.

Now that you mention it, I don't think my IDE (code-server) highlights these either.  I'll see if I can track down the discrepancy.  The storybook build is a little more forgiving, maybe that configuration is what's getting picked up by the IDE linter.

**Comment by indraraj** [comment]
Updated the card with a Preview label and cube icons on the top
image

**Comment by jgiardino** [comment]
Last week, I created this design task which is targeted for some time after LA: https://issues.redhat.com/browse/MGDCTRS-1710 

There are a few notes captured there that could be applied to your current PR and might be easy changes, but I'm also fine with your PR as-is. The updates that I think would be minor are 1) keeping the Owner property for customer owned namespaces as well as preview namespaces, and 2) removing the Created property.

**Comment by prapthik** [comment]
@indraraj This looks good to me.

**Comment by indraraj** [comment]
> Last week, I created this design task which is targeted for some time after LA: https://issues.redhat.com/browse/MGDCTRS-1710
> 
> There are a few notes captured there that could be applied to your current PR and might be easy changes, but I'm also fine with your PR as-is. The updates that I think would be minor are 1) keeping the Owner property for customer owned namespaces as well as preview namespaces, and 2) removing the Created property.

Thanks, I have updated the PR with suggestions

**Pull request merged**